### ISSUE TITLE
Append parquet files to minimize use of RAM by Redis

### ISSUE THREAD

**Issue by yohplala** [bmoscon/cryptostore#108]
Hi,

This feature request is also related to #19 (_too much files with storage=parquet, each interval for each backend separately is needed_) and is a way to solve, while at the same time addressing another problem #19 is not solving: use of RAM by Redis.

So, my trouble: I am renting a VPS at Hetzner, it has 2GB of RAM and 20GB of disk space.
I am recording about 250 channels (a channel being an exchange/data_type/pair), dumping data from Redis to parquet file every 1 hour.

Guess what?
```bash
Out of memory: Killed process 50483 (redis-server): total-vm:763916kB, anon-rss:653552kB,file-rs:0kB, shmem-rss:0kB, UID:110 pgtables:1392kB oom_score_adj:0
```
I understand I am out of memory.

So, the proposal here is to add in the config file a `append_counter` parameter to the config file in `parquet` section.

```yaml
parquet:
    del_file: true
    append_counter: 3
```

This `append_counter` will tell `cryptostore` to append data to the existing parquet file 3 times before starting a new parquet file.
Parquet file will only be uploaded to one of the data storage service (GC, GD or S3) and deleted once this 3 times have ellapsed.

So, if you have a `storage_interval` of 1min and a `append_counter` of 3 times. 
So every hour, you will have 60/(3+1) = 15 parquet files.
This answers to #19 I think.

This also answers my own problem:
- I want to have only one parquet file every hour (this is due to a Google Drive limit: I can have only 400 000 files per project. With about 250 channels, recorded once per hour, I will reach 400 000 files in about 3 to 4 months, which gives me plenty of time to process the data, to aggregate it to one file per month then),
- but because my RAM is limited, I need to dump RAM from redis much more frequently,

This way, I am able to benefit my 20GB of disk space while not requiring a lot of RAM.
:)

Googling appending of parquet files, I found [this SO](https://stackoverflow.com/questions/47113813/using-pyarrow-how-do-you-append-to-parquet-file) topic which would tend to show that parquet files can be appended.

@bmoscon 
Bryant, if you are ok with that, I will start this week-end to see how implementing this.

Bests,

**Comment by yohplala** [comment]
Hi,
I confirm I clarified most of the logic this evening and started the implementation.
Target to have a 1st code by tomorrow evening that I will check on Sunday evening.
Bests,

**Comment by yohplala** [comment]
Implemented in PR #110

**Comment by yohplala** [comment]
Closing / Implemented in PR #111
